TelemetryManager/Src/TMCircularBuffer.cpp

```diff
@@ -1,30 +1,56 @@
 
-#include "CircularBuffer.hpp"
+#include "TMCircularBuffer.hpp"
 
-TMCircularBuffer::TMCircularBuffer(uint8_t* buf, uint16_t size): CircularBuffer(buf, size) {
+TMCircularBuffer::TMCircularBuffer(uint8_t* buf, uint16_t size): CircularBuffer(*buf, size) {
     // Constructor
 }
 
-TMCircularBuffer::TMCircularBuffer(CircularBuffer buf): CircularBuffer(buf) {
+TMCircularBuffer::TMCircularBuffer(CircularBuffer* buf): CircularBuffer(*buf) {
```

Comment:
I don't see there is a constructor in the parent class that does `CircularBuffer::CircularBuffer(CircularBuffer* buf) `

Reply:
This was my fault. I asked Rahul to make a last-minute change.

Reply:
this function is still questionable? Can you confirm that this will work?

Reply:
@RRahul2004 are you able to do some manual testing to confirm this works?